kernel/thread.c

```diff
@@ -271,6 +282,18 @@ static inline size_t adjust_stack_size(size_t stack_size)
 	return stack_size - fuzz;
 }
 
+int stack_adjust_init(struct device *unused)
+{
+	/* Some random sources use kernel objects which can't be used until
+	 * after the main/idle threads start. Use the early boot version
+	 * before this happens.
+	 */
+	stack_adjust_initialized = 1;
+	return 0;
+}
+
+SYS_INIT(stack_adjust_init, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
```

Comment:
This mechanism seems needlessly complicated.  "stack_adjust_init" isn't a subsystem, it's a marker that says when a subsystem level has been initialized.

So why not put this assignment into init.c right after _sys_device_do_config_level(_SYS_INIT_LEVEL_POST_KERNEL)?  Simpler, smaller, clearer.

Longer term, we'll want to farm this out to the entroy driver and let it "register" or something at whatever point it gets initialized, and this will no doubt go away anyway.  So IMHO pick the simpler choice.

Reply:
good idea